File: TestSample/php/src/World.php
<?php
declare(strict_types=1);

namespace HelloWorld;

final class World
{
    /** @var Person[] */
    private array $people = [];

    public function add(Person $person): void
    {
        $this->people[] = $person;
    }

    /** @return string[] */
    public function helloAll(): array
    {
        $lines = [];
        foreach ($this->people as $p) {
            $lines[] = $p->hello();
        }
        return $lines;
    }
}

File: TestSample/csharp/src/World.cs
using System.Collections.Generic;

namespace HelloWorld;

public sealed class World
{
    private readonly List<Person> _people = new();

    public void Add(Person person) => _people.Add(person);

    public IEnumerable<string> HelloAll()
    {
        foreach (var person in _people)
        {
            yield return person.Hello();
        }
    }
}
